packages/swapper/src/swappers/zrx/ZrxApprovalNeeded/ZrxApprovalNeeded.ts

```diff
@@ -30,17 +30,18 @@ export async function ZrxApprovalNeeded(
   if (!quote.sellAsset.tokenId || !quote.allowanceContract) {
     throw new SwapError('ZrxApprovalNeeded - tokenId and allowanceTarget are required')
   }
+
   const allowanceResult = await getERC20Allowance({
     web3,
     erc20AllowanceAbi,
     tokenId: quote.sellAsset.tokenId,
     spenderAddress: quote.allowanceContract,
     ownerAddress: receiveAddress
   })
-  const allowanceOnChain = new BigNumber(allowanceResult || '0')
+  const allowanceOnChain = bnOrZero(allowanceResult)
 
   return {
-    approvalNeeded: allowanceOnChain.lt(new BigNumber(quote.sellAmount || 1)),
+    approvalNeeded: allowanceOnChain.lt(bnOrZero(quote.sellAmount || 1)),
```

Comment:
```suggestion
    approvalNeeded: allowanceOnChain.gte(bnOrZero(quote.sellAmount)),
```

allowance should be >= sell amount?